java/org/tartarus/snowball/SnowballProgram.java

```diff
@@ -1,52 +1,82 @@
 
 package org.tartarus.snowball;
-import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.UndeclaredThrowableException;
 import java.io.Serializable;
+import java.util.Arrays;
 
+/**
+ * Base class for a snowball stemmer
+ */
 public class SnowballProgram implements Serializable {
     protected SnowballProgram()
     {
-	current = new StringBuilder();
-	init();
+	setCurrent("");
     }
 
     static final long serialVersionUID = 2016072500L;
 
-    private void init() {
-	cursor = 0;
-	limit = current.length();
-	limit_backward = 0;
-	bra = cursor;
-	ket = limit;
-    }
-
     /**
      * Set the current string.
      */
     public void setCurrent(String value)
     {
-        // Make a new StringBuilder.  If we reuse the old one, and a user of
-        // the library keeps a reference to the buffer returned (for example,
-        // by converting it to a String in a way which doesn't force a copy),
-        // the buffer size will not decrease, and we will risk wasting a large
-        // amount of memory.
-        // Thanks to Wolfram Esser for spotting this problem.
-        current = new StringBuilder(value);
-	init();
+	setCurrent(value.toCharArray(), value.length());
     }
 
     /**
      * Get the current string.
      */
     public String getCurrent()
     {
-        return current.toString();
+        return new String(current, 0, limit);
```

Comment:
I think this `limit` should be `length`.

Reply:
thank you!